wire-schema/src/main/java/com/squareup/wire/schema/Loader.java

```diff
@@ -28,86 +30,58 @@
 
 /** Load proto files and their transitive dependencies, parse them, and link them together. */
 public final class Loader {
-  private final IO io;
-  private final Map<String, ProtoFile> loaded = new LinkedHashMap<>();
+  private final List<Path> dirs;
+  private final Map<Path, ProtoFile> loaded = new LinkedHashMap<>();
 
-  public Loader(IO io) {
-    this.io = io;
+  public Loader(List<Path> dirs) {
+    this.dirs = dirs;
   }
 
-  /** Returns a loader that loads all files from {@code base}. */
-  public static Loader forSearchPaths(final List<String> paths) {
-    IO io = new IO() {
-      @Override public Location locate(String file) throws IOException {
-        for (String path : paths) {
-          File f = new File(path, file);
-          if (f.exists()) {
-            return Location.get(path, file);
-          }
-        }
-
-        throw new IOException("Could not locate " + file + " within any of the following: "
-                + Joiner.on(", ").join(paths) + ".");
-      }
-
-      @Override public Source open(Location location) throws IOException {
-        return Okio.source(new File(location.base(), location.path()));
-      }
-
-      @Override public String toString() {
-        return String.format("Loader(%s)", Joiner.on(":").join(paths));
-      }
-    };
-    return new Loader(io);
+  /** Returns a loader that loads all files from {@code paths}. */
+  public static Loader forSearchPaths(final List<Path> paths) {
+    return new Loader(paths);
   }
 
-  public Schema load(Iterable<String> sourceFiles) throws IOException {
-    for (String path : sourceFiles) {
+  public Schema load(Iterable<Path> sourceFiles) throws IOException {
+    for (Path path : sourceFiles) {
       load(path);
     }
     return new Linker(loaded.values()).link();
   }
 
-  private void load(String path) throws IOException {
+  private void load(Path path) throws IOException {
     if (loaded.containsKey(path)) {
       return;
     }
 
-    Location location;
-    try {
-      location = io.locate(path);
-    } catch (IOException e) {
-      throw new IOException("Failed to locate " + path + " with " + io, e);
+    ProtoFileElement element = null;
+    for (Path dir : dirs) {
```

Comment:
I renamed to `protoDirs` and `protoFiles` for now. We can bikeshed the names in the follow-up where we'll expose these names in public API.